jdaviz/core/template_mixin.py

```diff
@@ -209,19 +251,20 @@ def _update_spectral_subset(self, msg):
 
         if msg.attribute == 'subset_state' and msg.subset.label == self.selected:
             # updated the currently selected subset
-            self._clear_cache("selected_obj")
+            self._clear_cache("selected_obj", "selected_item")
             self._update_has_subregions()
 
     def _selected_changed(self, event):
         if event['new'] not in self.labels:
-            self.selected = self.labels[0]
+            if len(self.labels):
+                self.selected = self.labels[0]
             raise ValueError(f"{event['new']} not one of {self.labels}")
```

Comment:
this is where the aperture photometry tests are currently tripping (see the comment below on accessing the subset object as well).  The aperture photometry test might need to be updated to remove the checks for invalid subset entries as this now raises an error (although we should make sure it doesn't put the plugin in a state it can't recover from after the error)